### PR TITLE
Add vuid 06433, Validate LocalSizeId execution mode

### PR DESCRIPTION
**Pull request by ziga-lunarg**

### CODE REVIEW

File: layers/shader_validation.cpp [hunk 1]
```diff
@@ -2081,6 +2081,10 @@ bool CoreChecks::ValidateExecutionModes(SHADER_MODULE_STATE const *module_state,
                         skip |= LogError(device, "VUID-RuntimeSpirv-LocalSizeId-06434",
                                          "LocalSizeId execution mode used but maintenance4 feature not enabled");
                     }
+                    if (!IsExtEnabled(device_extensions.vk_khr_maintenance4)) {
+                        skip |= LogError(device, "VUID-RuntimeSpirv-LocalSizeId-06433",
+                                         "LocalSizeId execution mode used but maintenance4 extension is not enabled");
```
Comment (sjfricke) [inline layers/shader_validation.cpp:2086]: I would slightly rephrase this to say

> LocalSizeId execution mode used but maintenance4 extension is not enabled and using Vulkan 1.2 or less

Comment (ncesario-lunarg) [reply layers/shader_validation.cpp:2086]: +1. Looks like an `|| api_version < VK_API_VERSION_1_3` check would also make sense?

Comment (ziga-lunarg) [reply layers/shader_validation.cpp:2086]: I don't think it's needed, if VK_API_VERSION is 1.3 then `device_extensions.vk_khr_maintenance4` will be `kEnabledByApiLevel`.

**Review by ncesario-lunarg — APPROVED** [review]
